### PR TITLE
Inline AddExchanges#Context

### PR DESCRIPTION
**Pull request by kokosing**
Inline AddExchanges#Context

### CODE REVIEW

**Comment by kokosing** [comment]
Please do not review unless Travis is green

File: presto-main/src/main/java/com/facebook/presto/sql/planner/optimizations/AddExchanges.java [hunk 1]
```diff
@@ -593,7 +583,7 @@ private PlanWithProperties planTableScan(TableScanNode node, Expression predicat
                     session,
                     symbolAllocator.getTypes(),
                     node.getAssignments(),
-                    filterConjuncts(constraint, conjunct -> !referencesAny(conjunct, context.getCorrelations())));
```
Comment (findepi) [inline presto-main/src/main/java/com/facebook/presto/sql/planner/optimizations/AddExchanges.java]: It looks like they were "used", but they were always empty. At this stage there should be no correlations left?

(Also please add some explanation to commit message regarding this.)

Comment (kokosing) [reply presto-main/src/main/java/com/facebook/presto/sql/planner/optimizations/AddExchanges.java]: > At this stage there should be no correlations left?

I forgot to check why they was added.

Comment (sopel39) [reply presto-main/src/main/java/com/facebook/presto/sql/planner/optimizations/AddExchanges.java]: I think when some subquery is not translated by planner, then correlations can still be here. Therefore previously this wouldn't fail, but rather some validator at the end of planning.

As we are with topic of validators. @kokosing could you tell me about relationship between:
1. `com/facebook/presto/sql/planner/sanity/PlanSanityChecker.java:38`
2. `com/facebook/presto/sql/planner/sanity/PlanSanityChecker.java:46`
3. `com/facebook/presto/sql/planner/PlanOptimizers.java:299`, `CheckSubqueryNodesAreRewritten`

It seems that either 1)2) or 3) is unnecessary. Do you recall why we have such double check?

Comment (kokosing) [reply presto-main/src/main/java/com/facebook/presto/sql/planner/optimizations/AddExchanges.java]: > I think when some subquery is not translated by planner, then correlations can still be here (at last some time ago it was like that).

That would be tree if not existence of `CheckSubqueryNodesAreRewritten`. This is an "optimizer" which enforce no subquery related nodes are in the plan. Further optimizers did not work well with subquery related nodes. See `e9c459a3d8`.

1.2 are about checking if subquery related `AST` entities were rewritten to plan entities, while 3 check if subquery related plan entities are rewritten to generic plan nodes.

**Pull request merged**